src/SkeletalTrapezoidationEdge.h

```diff
@@ -39,11 +70,53 @@ class SkeletalTrapezoidationEdge
     {
         return is_marked >= 0;
     }
+
+    bool hasTransitions(bool ignore_empty = false) const
+    {
+        return transitions_.use_count() > 0 && (ignore_empty || ! transitions_.lock()->empty());
+    }
+    void initTransitions(std::shared_ptr<std::list<TransitionMiddle>> storage)
+    {
+        transitions_ = storage;
+    }
+    std::shared_ptr<std::list<TransitionMiddle>> transitions()
+    {
+        return transitions_.lock();
+    }
+
+    bool hasTransitionEnds(bool ignore_empty = false) const
+    {
+        return transition_ends_.use_count() > 0 && (ignore_empty || ! transition_ends_.lock()->empty());
+    }
+    void initTransitionEnds(std::shared_ptr<std::list<TransitionEnd>> storage)
+    {
+        transition_ends_ = storage;
+    }
+    std::shared_ptr<std::list<TransitionEnd>> transition_ends()
+    {
+        return transition_ends_.lock();
+    }
+
+    bool hasExtrusionJunctions(bool ignore_empty = false) const
+    {
+        return extrusion_juntions_.use_count() > 0 && (ignore_empty || ! extrusion_juntions_.lock()->empty());
+    }
+    void initExtrusionJunctions(std::shared_ptr<std::vector<ExtrusionJunction>> storage)
+    {
+        extrusion_juntions_ = storage;
+    }
+    std::shared_ptr<std::vector<ExtrusionJunction>> extrusion_juntions()
+    {
+        return extrusion_juntions_.lock();
+    }
+
 private:
     int_least8_t is_marked; //! whether the edge is significant; whether the source segments have a sharp angle; -1 is unknown
-};
-
 
+    std::weak_ptr<std::list<TransitionMiddle>> transitions_;
```

Comment:
Nitpicking, but we usually use _transitions to mark something as protected (or well, it's because of python). But should we in the case of C++ even make a distinction? I don't think we do that in the rest of the code as far as I know.

Reply:
I used this because the function names would conflict (that is also why the function names are this way ... because I would've liked to use them as fields), not an issue anymore when I rename the functions to what you suggested.

I never know which side the `_` goes on anymore, since I've been forced to use two styles that used opposite endings a number of times :-p